File: app/models/crontab.rb
class Crontab < ApplicationRecord

  def self.start( cron_name )
    r = Crontab.where( cron_name: cron_name ).first_or_create!

    if r.status
      Banner.p 'Process actually running. Exiting'
      exit
    end

    r.update!( status: true, updated_at: Time.now )
  end

  def self.stop( cron_name )
    Crontab.where( cron_name: cron_name ).update_all( status: false, updated_at: Time.now )
  end

end

File: evebs/models/crontab.py
from datetime import datetime
from evebs.extensions import db


class Crontab(db.Model):
    __tablename__ = 'crontabs'

    id = db.Column(db.BigInteger, primary_key=True)
    cron_name = db.Column(db.String, nullable=False)
    status = db.Column(db.Boolean, default=False, nullable=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    @classmethod
    def start(cls, cron_name):
        import os
        if os.environ.get('FLASK_ENV') == 'development':
            return
        record = cls.query.filter_by(cron_name=str(cron_name)).first()
        if not record:
            record = cls(cron_name=str(cron_name), status=False)
            db.session.add(record)
        if record.status:
            print('Process actually running. Exiting')
            import sys
            sys.exit(0)
        record.status = True
        record.updated_at = datetime.utcnow()
        db.session.commit()

    @classmethod
    def stop(cls, cron_name):
        cls.query.filter_by(cron_name=str(cron_name)).update(
            {'status': False, 'updated_at': datetime.utcnow()}
        )
        db.session.commit()
